### PR TITLE
Meta: Add cache duration setter

### PR DESCRIPTION
**Pull request by facuspagnuolo**

### CODE REVIEW

File: pkg/pool-stable/test/meta/MetaStablePool.test.ts [hunk 1]
```diff
@@ -583,104 +596,129 @@ describe('MetaStablePool', function () {
...
+            updatePriceRateCache();
+            itAdaptsTheScalingFactorsCorrectly();
+          });
 
-            expect(newPriceRate0).to.be.gt(oldPriceRate0);
-            expect(newPriceRate1).to.be.gt(oldPriceRate1);
+          context('when it passed the previous duration', () => {
+            mockNewRatesAndAdvanceTime(MINUTE * 2);
+            updatePriceRateCache();
+            itAdaptsTheScalingFactorsCorrectly();
```
Comment (nventuro) [inline pkg/pool-stable/test/meta/MetaStablePool.test.ts:707]: It seems like these tests don't really test the new duration is in effect?

File: pkg/pool-stable/test/meta/MetaStablePool.test.ts [hunk 2]
```diff
@@ -583,104 +596,129 @@ describe('MetaStablePool', function () {
...
+          });
+        });
```
Comment (nventuro) [inline pkg/pool-stable/test/meta/MetaStablePool.test.ts:709]: It seems like these tests don't really test the new duration is in effect?

**Review by nventuro — COMMENTED** [review]
Fantastic, very clean :ok_hand:

File: pkg/pool-stable/test/meta/MetaStablePool.test.ts (hunk 2 above)
Comment (facuspagnuolo) [reply pkg/pool-stable/test/meta/MetaStablePool.test.ts:709]: It does, note that `itAdaptsTheScalingFactorsCorrectly` is querying the provider directly and these where updated right before the assertion, then the current scaling factors are being tested against the latest values coming from the providers

**Review by nventuro — APPROVED** [review]

**Pull request merged**